tests/rpmbuild.at

```diff
@@ -2983,6 +2986,24 @@ runroot rpm -ql /build/RPMS/noarch/dynamic-docs-1.0-1.noarch.rpm
 [])
 RPMTEST_CLEANUP
 
+# ------------------------------
+# Check if dynamic spec generation works for main package, too
+# Check for failure as feature is disabled. Remove test when enabled
+AT_SETUP([rpmbuild with dynamic spec generation for main package])
+AT_KEYWORDS([build])
+RPMDB_INIT
+RPMTEST_CHECK([
+
+runroot rpmbuild --define "_prefix /usr/local" -D "FULLDYNAMIC 1" -ba /data/SPECS/dynamic.spec
```

Comment:
But this is still testing -ba which wasn't the problem to begin with, -bs is:

> rpmbuild -D "FULLDYNAMIC 1" -bs ../tests/data/SPECS/dynamic.spec; echo $?
[...]
error: Summary field must be present in package: dynamic
error: License field must be present in package: dynamic
Wrote: /home/pmatilai/rpmbuild/SRPMS/dynamic-1.0-1.src.rpm
0

This patch does fix that, but test for it too. The cost here is disabling the previously working -ba case but if this is the best we can do then it's the best we can do.

Reply:
Well, the -ba thing no longer working is very much on purpose and not an unwanted side effect. We want a spec file to either be legal and producing binary and source packages or it to fail - no matter what operation one tries. As this now fails at the parse stage -bs and -ba don't make a difference. But I can run the same test with -bs, too.

Reply:
Here you go!

Reply:
My point is that -bs cannot like ever work on a highly dynamic package where you just have to run the build to get the results, whereas -ba *could* work (if rpmbuild internals permit :smile: ). So it's useful to separate the cases.